src/redux/books/books.js

```diff
@@ -0,0 +1,29 @@
+const ADD_BOOK = 'book-store/books/ADD_BOOK';
+const REMOVE_BOOK = 'book-store/books/REMOVE_BOOK';
+
+const bookReducer = (state = { arr: [] }, action) => {
+  switch (action.type) {
+    case 'ADD_BOOK':
+      return [...state, action.payload];
```

Comment:
- [ ] In redux, your initial state is the default value that you pass before your first action is triggered. You have set your initial state to an object with the key `arr` that has a value of an empty array. When you try to run the first action type of `ADD_BOOK`, it would return an error that says `state is not iterable`. I would advise you to make your state an empty array and not an object that contains an array and then update the logic for case REMOVE_BOOK to be `state.filter((book) => book.id !== action.payload);` 😎 💻